app/controllers/api/v1/participant_declarations_controller.rb

```diff
@@ -11,7 +11,14 @@ class ParticipantDeclarationsController < Api::ApiController
       include ApiFilterValidation
 
       def create
-        service = RecordDeclaration.new({ cpd_lead_provider: }.merge(permitted_params["attributes"] || {}))
+        attributes = permitted_params["attributes"] || {}
+
+        if FeatureFlag.active?(:disable_npq_endpoints) && attributes[:course_identifier].to_s.starts_with?("npq-")
```

Comment:
I understand why we might have this at the controller level, is there an argument of having this check at the record service level instead 🤔

Reply:
As we can identify the NPQ declarations, I think it's best to return as soon as possible; that way, we don't interfere with the service classes.

Reply:
The feature check is now done in `RecordDeclaration` service class